sql/hive/src/main/scala/org/apache/spark/sql/hive/HiveUtils.scala

```diff
@@ -66,6 +66,12 @@ private[spark] object HiveUtils extends Logging {
     .stringConf
     .createWithDefault(builtinHiveVersion)
 
+  // A deprecated config which is only used to provide a default value, in case some existing
+  // applications depend on this config, e.g. Spark SQL ODBC driver.
```

Comment:
i guess the jdbc driver do not depend on it. it used to only return `1.2.1`, but i think that this server may work fine with other hive versions

Reply:
it's ODBC. Actually it's something out of Spark's control, we must be careful and avoid behavior changes, like `conf.get("spark.sql.hive.version")` should still return `1.2.1` instead of null.

Reply:
if Thrift Server only works for 1.2.1，I guess we don’t need an isolated classload for its hive client like SparkSQLCLIDriver